### PR TITLE
Add a function to cleanup tombstones.

### PR DESCRIPTION
**Pull request by Gouthamve**
Still not happy with the interfaces. But let me know if a function like this makes sense.

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/tsdb/210)

### CODE REVIEW

**Comment by Gouthamve** [comment]
Tests will fail until #209 is merged

File: db.go [hunk 1]
```diff
@@ -686,6 +692,37 @@ func (db *DB) Delete(mint, maxt int64, ms ...labels.Matcher) error {
...
+	deleted := []string{}
+	for _, b := range blocks {
+		ok, err := b.CleanTombstones(db.Dir(), db.compactor)
+		if err != nil {
+			return errors.Wrapf(err, "clean tombstones: %s", b.Dir())
+		}
+
+		if ok {
+			deleted = append(deleted, b.Dir())
+		}
```
Comment (fabxc) [inline db.go]: It looks like we just delete any block the we applied tombstones to?

Comment (Gouthamve) [reply db.go]: Yes, thats because we are compacting the data into a new block.

Comment (fabxc) [reply db.go]: Oh right, ULID changes – makes sense

**Comment by fabxc** [comment]
👍

**Pull request merged**